python/lsst/ts/mtaos/model.py

```diff
@@ -1009,6 +1023,90 @@ def _generate_pipetask_command(
 
         return run_pipetask_cmd
 
+    async def _poll_butler_outputs(
+        self,
+        run_name: str,
+        visit_id: int,
+        instrument: str,
+        timeout: int = 60,
+        poll_interval: int = 5,
+    ) -> list:
+        """
+        Poll the Butler for the outputs of a given run
+        and visit id, with a timeout.
+
+        Parameters
+        ----------
+        run_name : `str`
+            Name of the run.
+        visit_id : `int`
+            Id of the visit.
+        instrument : `str`
+            Camera used to take the data.
+        timeout : `int`, optional
+            Maximum time to wait for the outputs (in seconds).
+        poll_interval : `int`, optional
+            How often to poll for the data (in seconds).
+
+        Returns
+        -------
+        `list`
+            List of wavefront errors from the Butler.
+
+        Raises
+        ------
+        TimeoutError
+            If the dataset is not available within the specified timeout.
+        """
+        self.log.debug("Polling butler for WEP outputs.")
+
+        butler = dafButler.Butler(self.data_path)
+        start_time = time.time()
+
+        while True:
+            elapsed_time = time.time() - start_time
+
+            datasetRefs = list(
+                butler.registry.queryDatasets(
+                    datasetType="postISRCCD", collections=[run_name]
+                )
+            )
+            if datasetRefs:
+                self.log.debug(f"Found dataset references: {datasetRefs}")
+                break
+
+            data_ids = butler.registry.queryDatasets(
```

Comment:
Is this in the right place?

Reply:
Well I think so, because what we care is not the postISRCCD but the zernike_table_name output no? So that's the one we need to wait for no?